src/renderers/dom/shared/__tests__/ReactDOMServerIntegration-test.js

```diff
@@ -2582,34 +2582,4 @@ describe('ReactDOMServerIntegration', () => {
         <div dangerouslySetInnerHTML={{__html: "<span id='child2'/>"}} />,
       ));
   });
-
-  describe('dynamic injection', () => {
-    beforeEach(() => {
-      // HACK: we reset modules several times during the test which breaks
-      // dynamic injection. So we resort to telling resetModules() to run
-      // our custom init code every time after resetting. We could have a nicer
-      // way to do this, but this is the only test that needs it, and it will
-      // be removed anyway when we switch to static injection.
-      onAfterResetModules = () => {
-        const DOMProperty = require('DOMProperty');
-        DOMProperty.injection.injectDOMPropertyConfig({
-          isCustomAttribute: function(name) {
-            return name.indexOf('foo-') === 0;
-          },
-          Properties: {foobar: null},
-        });
-      };
-      resetModules();
-    });
-
-    afterEach(() => {
-      onAfterResetModules = null;
-    });
-
-    itRenders('injected attributes', async render => {
```

Comment:
Here's my (failing) test case:

```javascript
  describe.only('dynamic injection', () => {
    beforeEach(() => {
      // HACK: we reset modules several times during the test which breaks
      // dynamic injection. So we resort to telling resetModules() to run
      // our custom init code every time after resetting. We could have a nicer
      // way to do this, but this is the only test that needs it, and it will
      // be removed anyway when we switch to static injection.
      onAfterResetModules = () => {
        const DOMProperty = require('DOMProperty');
        DOMProperty.injection.injectDOMPropertyConfig({
          Properties: {foobar: DOMProperty.injection.MUST_USE_PROPERTY},
        });
      };
      resetModules();
    });

    afterEach(() => {
      onAfterResetModules = null;
    });

    itRenders('injected attributes', async render => {
      const test = {}
      const e = await render(<div foobar={test} />, 0);

      expect(e.foobar).toEqual(test)
    });
  });
```

What is the best injection to test with minimal friction, or should I start thinking about this case (server rendering an object attribute).